blenderproc/python/writer/BopWriterUtility.py

```diff
@@ -131,14 +132,41 @@ def write_bop(output_dir: str, target_objects: Optional[List[MeshObject]] = None
         chunk_dir_ids = [d.split('/')[-1] for d in chunk_dirs]
         chunk_dirs = chunk_dirs[chunk_dir_ids.index(f"{starting_chunk_id:06d}"):]
 
+        # convert all objects to trimesh objects
+        trimesh_objects = {}
+        for obj in dataset_objects:
+            trimesh_obj = obj.mesh_as_trimesh()
+            if m2mm:
+                trimesh_obj.apply_scale(scaling=0.001)
+            trimesh_objects[obj.get_cp('category_id')] = pyrender.Mesh.from_trimesh(mesh=trimesh_obj)
```

Comment:
Is this safe? What if there are objects with the same category id?

Reply:
i think if the objects are loaded with `load_bop_objs` then this is a unique mapping from object models (i.e., if an instance exists twice, the model exists twice too). See [here](https://github.com/DLR-RM/BlenderProc/blob/762bd65e6f37b4c8e00f8d4e0826b4575f005fea/blenderproc/python/loader/BopLoader.py#L349); and the corresponding `gt_id` in the annotations also references that ([here](https://github.com/DLR-RM/BlenderProc/blob/762bd65e6f37b4c8e00f8d4e0826b4575f005fea/blenderproc/python/writer/BopWriterUtility.py#L281))

Reply:
Keep in mind that there can be multiple objects from one category id in BOP and that the bopwriter should also work without the boploader :)

Reply:
It's actually fine, but could be more efficient 
https://github.com/DLR-RM/BlenderProc/pull/881#discussion_r1186861587

Reply:
But its not always the case that objects with the same category id have the same mesh, right?
Sometimes there are different object instances that share the same category id

Reply:
> But its not always the case that objects with the same category id have the same mesh, right?
Sometimes there are different object instances that share the same category id

For BOP datasets `category_id` represents the `obj_id`, i.e. a specific instance. I think this was to make it compatible to other components such as the coco_writer.